core/src/main/java/com/graphhopper/matrix/algorithm/MatrixAlgorithmFactory.java

```diff
@@ -0,0 +1,20 @@
+package com.graphhopper.matrix.algorithm;
+
+import com.graphhopper.routing.AlgorithmOptions;
+import com.graphhopper.storage.Graph;
+
+/**
+ * Builds MatrixAlgorithms according to specified parameters.
+ *
+ * @author Pascal Büttiker
+ */
+public interface MatrixAlgorithmFactory {
+
+    /**
+     * Creates a MatrixAlgorithm instance using the given options
+     * @param g The Graph to run the algorithm on
+     * @param opts The options for the algorithm
+     */
+    MatrixAlgorithm build(Graph g, AlgorithmOptions opts );
```

Comment:
I would use the same naming `createAlgo` for now? (`RoutingAlgorithm createAlgo( Graph g, AlgorithmOptions opts );`)